GoogleForCreators/web-stories-wp
# Pre-Publish Checklist: Limit story poster check to 1 at a time

## Context

There were situations where we'd get two checklist issues on a poster image if the aspect ratio was off and the size was less than the recommended. It was redundant to see both at once. 

## Summary

Since https://github.com/google/web-stories-wp/issues/7744 was merged, it's now near impossible to make a poster image the wrong aspect ratio. ~I left that check for edge cases and added a further specification to make sure the height/width minimum is met before showing the aspect ratio issue.~ Combined the `storyAspectRatioCheck` with `storyPosterPortraitSizeCheck` so you only see the one check.


## Relevant Technical Choices

added this:
```
  featuredMedia?.height < FEATURED_MEDIA_RESOURCE_MIN_HEIGHT ||
  featuredMedia?.width < FEATURED_MEDIA_RESOURCE_MIN_WIDTH
``` 

to the storyPosterAspectRatio (now just `storyPosterSize`) check so that if either of these are found true the aspect ratio check early returns true and renders without needing to check on the aspect ratio meeting standards since it's too small. 

Combined tests.

## To-do



## User-facing changes

Should no longer see both of these checks (see screenshot) when a poster image is too small and has the wrong aspect ratio. 

![redundant checks](https://camo.githubusercontent.com/c210cf1aa88a36dacc8df7614ef30615c85b7a1c1a2b8ee3fdcd8d5d64f50100/68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3565396463643961396666356466646166666364663133322f35323837383135392d396338352d343133652d626338372d386463613431336633376132)

## Testing Instructions

Once you set a poster image, it's going to force cropping and resizing so it won't show the check (a good thing!). To force it, open a second window and go to your media in WordPress, edit the poster image instance and try making it real small or adjusting the aspect ratio and cropping. Then refresh the story and see the check show up.

### QA






This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #8129
